packages/pagination/src/styled/Pagination/StyledPage.ts

```diff
@@ -111,7 +114,7 @@ export const StyledPage = styled.li.attrs({
     font-weight: ${props => props.theme.fontWeights.semibold};
   }
 
-  ${props => colorStyles(props)};
+  ${props => !(props as any).disabled && colorStyles(props)};
```

Comment:
Can we take a different approach? Let's extract out a common set of styling values to a `StyledBase` (or some improved name) component. Then both `StyledPage` and `StyledCursor` would extend that base component. This should end up being more intuitive vs. the fighting resets that are happening between StyledPage and StyledCursor. Also, the `disabled` prop should be properly typed within the scope of `StyledCursor`.

Reply:
btw, if `disabled` is retained in `StyledPage` (for future use), we should take a different approach. Do not rely on UA styling for the color. Rather, all disabled-related styling should be applied at the end of the `colorStyles` function – probably to both the `:disabled` psuedo and the `[aria-disabled='true']` attributes. See https://github.com/zendeskgarden/react-components/blob/ba6076878aa729383b285e76ba37273054bfe215/packages/buttons/src/styled/StyledButton.ts#L132-L136 for details.

Reply:
> Can we take a different approach?

Yeah, I would love to! I originally had something like this in mind, but steered away from modifying `StyledPage`. 

I've moved all shared styling into a file called `StyledPageBase` (open to name ideas) and have both `StyledCursor` and `StyledBase` extending from the base styles. I've pushed up a new commit for this and updated the [demo](https://cursor-pagination.netlify.com/pagination/). Let me know what you think of these changes.